src/Database/Oracle/Simple/FromField.hs

```diff
@@ -62,14 +65,19 @@ instance (FromField a) => FromField (Maybe a) where
       then pure Nothing
       else Just <$> readDPIDataBuffer (fromField @a) ptr
 
--- instance FromField UTCTime where
---   fieldType Proxy = fieldType (Proxy @DPITimestamp)
---   fromField = do
---     DPITimestamp {..} <- fromField
---     let
---       yymmdd = fromGregorian (fromIntegral year) (fromIntegral month) (fromIntegral day)
---       hhmmss = 0
---     pure (UTCTime yymmdd hhmmss)
+instance FromField UTCTime where
+  fromField = dpiTimeStampToUTCTime <$> fromField
+
+dpiTimeStampToUTCTime :: DPITimestamp -> UTCTime
+dpiTimeStampToUTCTime dpi =
+  let
+    DPITimestamp {..} = dpiTimeStampToUTCDPITimeStamp dpi
+    local = LocalTime d tod
+    d = fromGregorian (fromIntegral year) (fromIntegral month) (fromIntegral day)
+    tod = TimeOfDay (fromIntegral hour) (fromIntegral minute) (fromIntegral second + picos)
+    picos = MkFixed (fromIntegral fsecond) :: Pico
```

Comment:
`fsecond` in the struct expects nanos, not picos
https://odpi-c.readthedocs.io/en/latest/structs/dpiTimestamp.html

I'd convert from nanos to picos here.